File: Java/Program/Employee.java
public class Employee extends Person { // Subclass from Person
    // Attributes
    private String employeeId;
    private String position;
    // Constructors
    public Employee() {}
    // Parameterized Constructor
    public Employee(String name, int age, String gender, String employeeId, String position) {
        super(name, age, gender);
        this.employeeId = employeeId;
        this.position = position;
    }
    // Getters & Setters
    public String getEmployeeId() { return employeeId; }
    public void setEmployeeId(String employeeId) { this.employeeId = employeeId; }

    public String getPosition() { return position; }
    public void setPosition(String position) { this.position = position; }
    // Method to print employee details
    public void printEmployee() {
        System.out.printf("%s\t%s\t%s\t%d\t%s%n",
            getEmployeeId(), getName(), getGender(), getAge(), getPosition());
    }
}

File: CPP/Program/Employee.cpp
#pragma once
#include <iostream>
#include <string>
#include "Person.cpp"

// This class represents an Employee, inheriting from Person
class Employee : public Person {
    // Attributes specific to Employee
    private:
        string employeeId;
        string position;

    public:
        // Default constructor
        Employee(){
        }

        // Parameterized constructor
        Employee(string name, int age, string gender, string employeeId, string position)
            : Person(name, age, gender) {
            this->employeeId = employeeId;
            this->position = position;
        }

        // Setters and Getters
        void setEmployeeId(string employeeId) {
            this->employeeId = employeeId;
        }

        string getEmployeeId() {
            return this->employeeId;
        }

        void setPosition(string position) {
            this->position = position;
        }

        string getPosition() {
            return this->position;
        }

        ~Employee(){
        }

        // For printing employee details
        void printEmployee() {
            cout << getEmployeeId() << "\t" << getName() << "\t" << getGender() << "\t" << getAge() << "\t" << getPosition() << endl;
        }
};